### PR TITLE
Migrate Triggers e2e test to kind

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Follows the guide in docs/kind-e2e.md with a couple of changes to match the current pipeline e2e tests:

1. Use latest builder image for the time being
2. Adds a `pipeline-kind-e2e: "true"` label that is used for podAntiAffinity - we don't want multiple kind jobs running on the same nodes.

Part of tektoncd/triggers#1475

/kind misc

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by dibyom** [comment]
/hold

I'll try to get https://github.com/tektoncd/triggers/pull/1477 in first

File: prow/config.yaml
Comment (abayer) [inline prow/config.yaml:1660]: We should probably rename this label, since its purpose is to ensure that kind e2e jobs each end up on their own node, regardless of project. But that doesn't need to be done here. =)

File: prow/config.yaml
Comment (abayer) [inline prow/config.yaml:1684]: This can stay the same - I should update the Pipeline jobs' images to not use `latest`, since the relevant changes have long since been merged.

**Review by abayer — APPROVED** [review]

**Comment by afrittoli** [comment]
> /hold
> 
> I'll try to get [tektoncd/triggers#1477](https://github.com/tektoncd/triggers/pull/1477) in first

This is merged now!

**Comment by dibyom** [comment]
> This is merged now!

Yes, keeping the hold for a few more days till we release the next Triggers version and then we'll switch to kind

File: prow/config.yaml
Comment (JeromeJu) [inline prow/config.yaml:1699]: Do we need "--e2e-env" here?

Comment (dibyom) [reply prow/config.yaml:1699]: Do you mean - --e2e-env vs --e2e-env? Or the e2e-env arg altogether?

Comment (JeromeJu) [reply prow/config.yaml:1699]: Oh I meant the quotation marks :)

**Comment by afrittoli** [comment]
@dibyom kind reminder about this 🙏

**Comment by dibyom** [comment]
> @dibyom kind reminder about this 🙏

I'll pick it back up this week!!

File: prow/config.yaml
Comment (dibyom) [reply prow/config.yaml:1699]: nice catch :)

File: prow/config.yaml
Comment (dibyom) [reply prow/config.yaml:1660]: I'll do that as part of https://github.com/tektoncd/plumbing/issues/1281

File: prow/config.yaml
Comment (dibyom) [reply prow/config.yaml:1684]: Done, I can change the other images as well in a separate PR

**Comment by dibyom** [comment]
@abayer @afrittoli ready for another look!

**Comment by dibyom** [comment]
just needs a LGTM now @afrittoli @vdemeester 🙏

**Review by vdemeester — APPROVED** [review]
/lgtm

**Pull request merged**
